core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/DefaultJsonContentTypeResolver.java

```diff
@@ -33,6 +33,7 @@ public DefaultJsonContentTypeResolver(String prefix) {
     @Override
     public String resolveContentType(AwsJsonProtocolMetadata protocolMetadata) {
         //Changing this to 'application/json' may break clients expecting 'application/x-amz-json-1.1'
-        return prefix + protocolMetadata.protocolVersion();
+        return protocolMetadata.contentTypeOverride() != null ? protocolMetadata.contentTypeOverride()
+                : prefix + protocolMetadata.protocolVersion();
```

Comment:
I don't like "overrides" being bound so late. It means we have to pipe a lot information through from the code generator, which adds a lot more cross-module API surface area. Can we just have this line be `return protocolMetadata.contentType();` and have the code generator define the content-type that's used?

We can have the default value be "prefix + protocolMetadata.protocolVersion()", but do that upstream in the protocol factory. I'd honestly prefer it to be defaulted in the code generator, but that would break people who just upgrade the JSON protocol module without updating their service clients (which we try not to do outside of minor version bumps).

Reply:
Agree.  Tried adding this changes in Codegen it self but there we have dependency of Protocol Version too and it make code complicated as as you said might break upgrade the JSON protocol module .
Moved this logic to BaseAwsJsonProtocolFactory.

Reply:
Moved the logic where the  check for customized ContentType is present  is moved to protocol factory.